cmd/config/init/unique_rollapp_id.go

```diff
@@ -1,36 +1,86 @@
 package initconfig
 
 import (
+	"encoding/json"
 	"fmt"
+	"github.com/dymensionxyz/roller/cmd/consts"
+	"github.com/dymensionxyz/roller/cmd/utils"
 	"github.com/dymensionxyz/roller/config"
 	"net/http"
+	"os/exec"
+	"regexp"
+	"strings"
 )
 
 // TODO(#150): roller should use RPC for queries instead of REST
 func IsRollappIDUnique(rollappID string, initConfig config.RollappConfig) (bool, error) {
-	url := initConfig.HubData.API_URL + "/dymensionxyz/dymension/rollapp/rollapp/" + rollappID
+	if initConfig.VMType == config.SDK_ROLLAPP {
+		url := initConfig.HubData.API_URL + "/dymensionxyz/dymension/rollapp/rollapp/" + rollappID
 
-	req, err := http.NewRequest("GET", url, nil)
-	if err != nil {
-		return false, err
+		req, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			return false, err
+		}
+
+		req.Header.Set("accept", "application/json")
+
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if err != nil {
+			return false, err
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode == 404 {
+			return true, nil
+		} else if resp.StatusCode == 200 {
+			return false, nil
+		} else {
+			return false, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+		}
+	} else {
+		return verifyUniqueEthIdentifier(rollappID, initConfig)
 	}
+}
 
-	req.Header.Set("accept", "application/json")
+type Rollapp struct {
+	ID string `json:"rollappId"`
+}
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+type RollappsListResponse struct {
+	Rollapps []Rollapp `json:"rollapp"`
+}
+
+func verifyUniqueEthIdentifier(rollappID string, rlpCfg config.RollappConfig) (bool, error) {
+	commonDymdFlags := utils.GetCommonDymdFlags(rlpCfg)
+	// TODO: Move the filtering by ethereum rollapp ID logic to the hub
+	args := []string{"q", "rollapp", "list", "--limit", "1000000"}
+	args = append(args, commonDymdFlags...)
+	listRollappCmd := exec.Command(consts.Executables.Dymension, args...)
+	out, err := utils.ExecBashCommandWithStdout(listRollappCmd)
+	if err != nil {
+		return false, err
+	}
+	rollappsListResponse := RollappsListResponse{}
+	err = json.Unmarshal(out.Bytes(), &rollappsListResponse)
 	if err != nil {
 		return false, err
 	}
-	defer resp.Body.Close()
+	for _, rollapp := range rollappsListResponse.Rollapps {
+		if getEthID(rollapp.ID) == getEthID(rollappID) {
+			return false, nil
+		}
+	}
+	return true, nil
+}
 
-	if resp.StatusCode == 404 {
-		return true, nil
-	} else if resp.StatusCode == 200 {
-		return false, nil
-	} else {
-		return false, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+func getEthID(rollappID string) string {
+	re := regexp.MustCompile(`_(\d+)-`)
```

Comment:
@ItayLevyOfficial the regex compilation should be done once, and not each loop iteration